lib/utils.js

```diff
@@ -50,7 +50,7 @@ exports.validateArg = function validateArg (args, config) {
         break;
 
       case Function:
-        if (toString.call(value) !== '[object Function]') {
+        if (typeof obj !== "undefined" && toString.call(value) !== '[object Function]') {
```

Comment:
As far as I can tell, passed functions are always callbacks, so they should be globally optional.   I have been unable to find any place where they are not, so that seems to me that would be writing code support for a case that doesn't exist.  I'm not intimately familiar with the module though, so please correct me if you know of one.

If there is one, IMO, simply checking for optionality before the call would require that I then pass a "fake" function to validateArg in order to get it to pass which seems very hacky.  If this is absolutely necessary I would add another item to the config called "OptionalFunction" and copy/paste the "Function" code except also allowing undefined, and then globally find/replace Function with OptionalFunction in every command config definition.

But again, unless a case is identified, this seems like code bloat for a situation that doesn't exist.